packages/ckeditor5-table/src/commands/mergecellscommand.js

```diff
@@ -57,40 +57,25 @@ export default class MergeCellsCommand extends Command {
 			updateNumericAttribute( 'colspan', mergeWidth, firstTableCell, writer );
 			updateNumericAttribute( 'rowspan', mergeHeight, firstTableCell, writer );
 
+			const emptyRows = [];
+
 			for ( const tableCell of selectedTableCells ) {
 				const tableRow = tableCell.parent;
+
 				mergeTableCells( tableCell, firstTableCell, writer );
-				removeRowIfEmpty( tableRow, writer );
+
+				if ( !tableRow.childCount ) {
+					emptyRows.push( table.getChildIndex( tableRow ) );
+				}
 			}
 
+			emptyRows.reverse().forEach( row => tableUtils.removeRows( table, { at: row, batch: writer.batch } ) );
```

Comment:
And since a `table` is only needed here maybe a wrap in `if( emptyRows.length )` ? But IDK for sure - this can be left as is.